src/registrar/management/commands/load_transition_domain.py

```diff
@@ -0,0 +1,463 @@
+                        total_new_entries += 1
+
+                        # DEBUG:
+                        if debug_on:
+                            logger.info(
+                                f"{termColors.OKCYAN} Adding entry {total_new_entries}: {newEntry} {termColors.ENDC}"  # noqa
+                            )
+                    except TransitionDomain.MultipleObjectsReturned:
+                        logger.info(
+                            f"""
+{termColors.FAIL}
+!!! ERROR: duplicate entries exist in the
+transtion_domain table for domain:
+{new_entry_domainName}
+----------TERMINATING----------"""
+                        )
+                        import sys
+
+                        sys.exit()
+
+                # DEBUG:
+                if debug_on or debug_max_entries_to_parse > 0:
+                    if (
+                        total_rows_parsed > debug_max_entries_to_parse
+                        and debug_max_entries_to_parse != 0
+                    ):
+                        logger.info(
+                            f"""{termColors.WARNING}
+                            ----PARSE LIMIT REACHED.  HALTING PARSER.----
+                            {termColors.ENDC}
+                            """
+                        )
+                        break
+
+        TransitionDomain.objects.bulk_create(to_create)
+
+        logger.info(
+            f"""{termColors.OKGREEN}
+
+        ============= FINISHED ===============
+        Created {total_new_entries} transition domain entries,
+        updated {total_updated_domain_entries} transition domain entries
+        {termColors.ENDC}
+        """
+        )
+
+        # Print a summary of findings (duplicate entries,
+        # missing data..etc.)
+        self.print_summary_duplications(
+            duplicate_domain_user_combos, duplicate_domains, users_without_email
+        )
+        self.print_summary_status_findings(domains_without_status, outlier_statuses)
+
+    def print_debug_mode_statements(self, debug_on, debug_max_entries_to_parse):
+        if debug_on:
+            logger.info(
+                f"""{termColors.OKCYAN}
+            ----------DEBUG MODE ON----------
+            Detailed print statements activated.
+            {termColors.ENDC}
+            """
+            )
+        if debug_max_entries_to_parse > 0:
+            logger.info(
+                f"""{termColors.OKCYAN}
+            ----------LIMITER ON----------
+            Parsing of entries will be limited to
+            {debug_max_entries_to_parse} lines per file.")
+            Detailed print statements activated.
+            {termColors.ENDC}
+            """
+            )
+
+    def get_domain_user_dict(self, domain_statuses_filename, sep):
+        """Creates a mapping of domain name -> status"""
+        # TODO: figure out latest status
+        domain_status_dictionary = defaultdict(str)
+        # NOTE: how to determine "most recent" status?
```

Comment:
Not sure, but I think you could maybe infer by its order in the array - so the last item in it? (wouldn't be reliable, though)

Reply:
More notes I will need to cleanup.  I believe the "most recent" status will be simply the last-to-be-parsed entry in the given Verisign file.  But I will double-check with Alysia.